### ISSUE TITLE
Menu sizing with Zenity v3.99 beta (upcoming v4.0)

### ISSUE THREAD

**Issue by arkhoss** [starcitizen-lug/lug-helper#67]
**Describe the bug**

Too small frame size of main menu.

**To Reproduce**
1. Run script `./lug-helper.sh`
2. Star Citizen LUG Helper screen loads
3. Scroll down to see hidden options

**Expected behavior**
Full menu without window resize or scrolling down

**Screenshots**

Actual

![menu](https://user-images.githubusercontent.com/28204104/272865478-8f621a13-b64c-4ccc-af8a-de663166c049.png)

Expected

![menuB](https://user-images.githubusercontent.com/28204104/272874478-9c0f6a57-70a1-4e1d-bbc4-1ce9419e8eee.png)

**OS Info:**

```
@fedora 
----------------- 
OS: Fedora Linux 38 (Workstation Edition) x86_64 
Kernel: 6.4.15-200.fc38.x86_64 
Uptime: 1 day, 10 hours, 52 mins 
Packages: 2279 (rpm), 10 (flatpak) 
Shell: bash 5.2.15 
Resolution: 1920x1080 
DE: GNOME 44.5 
WM: Mutter 
WM Theme: Adwaita 
Theme: Adwaita [GTK2/3] 
Icons: Adwaita [GTK2/3] 
Terminal: gnome-terminal 
CPU: AMD Ryzen 7 1800X (16) @ 3.600GHz 
GPU: AMD ATI Radeon RX 470/480/570/570X/580/580X/590 
Memory: 7674MiB / 32007MiB 
```

**Additional context**

If it is just me who see this as an issue, please disregard my MR. Thanks!

**Comment by arkhoss** [comment]
Submitted MR: https://github.com/starcitizen-lug/lug-helper/pull/68

A proposal to resolve this menu sizing issue.

**Comment by the-sane** [comment]
As mentioned in the PR, I don't think xrandr is going to be a good solution for this. To my knowledge, it won't work on Wayland, and I'd really prefer to avoid adding external dependencies.

I'm also not convinced this isn't a theming issue. That doesn't look like stock Adwaita to me and we have a lot of folks using resolutions of all sizes without having sizing issues that make it as unusable as in your screenshot.

Can you play around with theming/accessibility options and try to narrow this down a bit further? There must be something else going on here aside from screen resolution and if we can track that down, it may be an easier thing to detect and adjust for without adding external dependencies.

**Comment by arkhoss** [comment]
HI @the-sane ,

I'm using Wayland and it is stuck Adwaita in Fedora 38, no customization. 

```
$ echo "$XDG_SESSION_TYPE"
wayland
```

> Can you play around with theming/accessibility options and try to narrow this down a bit further?

I will check it out, and try to get some info about it. Thanks for the review!

**Comment by the-sane** [comment]
This is what stock Adwaita/Gnome looks like on my Arch machine:
![Screenshot from 2023-10-05 16-10-55](https://github.com/starcitizen-lug/lug-helper/assets/3657071/d01cc35c-4f1f-43f4-b005-d1c88c90ed2e)

Notice how I have a title bar and close/minimize buttons but you don't? That's what makes me wonder about a theming issue. On the other hand, I do recall that there was some kind of change coming down the pipes in a beta version of Zenity I think?

What version of Zenity does Fedora 38 package? I have v3.44.2

**Comment by arkhoss** [comment]
Hi @the-sane , 

it is a fresh install Fedora 38, and zenity info below:


```

$ sudo dnf info zenity
Installed Packages
Name         : zenity
Version      : 3.99.1
Release      : 1.fc38
Architecture : x86_64
Size         : 11 M
Source       : zenity-3.99.1-1.fc38.src.rpm
Repository   : @System
From repo    : updates
Summary      : Display dialog boxes from shell scripts
URL          : https://wiki.gnome.org/Projects/Zenity
License      : LGPL-2.1-or-later
Description  : Zenity lets you display Gtk+ dialog boxes from the command line and through
             : shell scripts. It is similar to gdialog, but is intended to be saner. It comes
             : from the same family as dialog, Xdialog, and cdialog.

```

Also, I've been looking for an alternative to xrandr and only alternative without external dependencies is with `cat` and this method relies on the presence of the frame buffer device and may not work on all systems.

```

cat /sys/class/graphics/fb0/modes | awk '{print $1}'

```

Any recommendations for checking out the resolution? I haven't found much around it.

**Comment by the-sane** [comment]
Ah, okay, so it's Zenity's fault and has nothing to do with screen resolution. Zenity v3.99 is a beta of the upcoming v4.0 release. The last time someone posted a screenshot, v3.91 seemed to show all the options correctly, so they made some change since then that broke it. It's possible that they change it again and resolve this before the final release.

We'll wait to see what happens when v4.0 gets released and then figure out what to do about it. We'll have to make sure we don't break backwards compatibility with previous zenity versions at that time.

I'm going to rename this issue and I'll use it as a reminder to figure something out when 4.0 is released.

**Comment by the-sane** [comment]
In the meantime btw, for anyone who needs to quickly fix this to make it usable, the quickest fix will be to open the script to [this line](https://github.com/starcitizen-lug/lug-helper/blob/beabf9c35c19d7010135a93ea7175bcbed1d7a5b/lug-helper.sh#L477) and change `--height="$menu_height"` to something like `--height="$((menu_height*3))"`, or `+400`, or whatever works for your situation.

**Comment by the-sane** [comment]
Further info on this: Zenity 4.0 has been released on Arch. After the holidays, I'll be working towards a solution that automatically accommodates current and previous versions without adding additional dependencies.

That's probably going to look like a version detection and two hardcoded height adjustments, similarly to how it's done currently. If anyone has other thoughts on how to dynamically adjust sizing for both xorg and wayland without adding additional dependencies, let me know.

**Comment by arkhoss** [comment]
Hi @the-sane ,

Maybe something like this #74  ? A proposal to resolve this menu sizing issue.

No use of external tools
Hardcoded values for height
Support old zenity versions < `3.44.x`
Support for versions above > `3.44.x` and less than `4`
Support for versions above 4

Tests:


Fedora 39 - zenity `4.0.0`

![Screenshot from 2023-12-22 03-51-52](https://github.com/starcitizen-lug/lug-helper/assets/28204104/349f238a-b65c-4fa2-9530-3deb5c5189a2)

Fedora 38 - zenity `3.44.0`

![Screenshot from 2023-12-22 04-48-09](https://github.com/starcitizen-lug/lug-helper/assets/28204104/1be62310-1412-4333-8c65-a899a13c63bd)

Manjaro 6.6.7-1 - zenity `3.44.2`

![Screenshot from 2023-12-22 04-49-59](https://github.com/starcitizen-lug/lug-helper/assets/28204104/10e8ce05-b6d3-4ac6-a771-b585e08a6f63)

Manjaro 6.6.7-1 - zenity `4.0.0`

![Screenshot from 2023-12-22 05-10-21](https://github.com/starcitizen-lug/lug-helper/assets/28204104/6819586e-33b5-42c5-8477-584bcb3c0a79)

**Comment by the-sane** [comment]
I solved this slightly differently in https://github.com/starcitizen-lug/lug-helper/commit/a9129db974de80ee850a883940acd28903f6cd18

I use a couple different height adjustments and a modifier to make the sizing as dynamic as possible without actually measuring the user's screen space. It's more complicated than I'd like, but it's the best way I've been able to come up with so far.

Basically, I take the number of menu items and multiply them by a fixed height value for each item. That value is different for zenity 3 and 4. Lastly, I add extra height for the title/description space that also varies depending on the version of zenity in use. And lastly, the max height cap also needs to be different.

Thanks for helping with some ideas though, I really appreciate it!
